### PR TITLE
Get rid of MonadRandom in VRF

### PR DESCRIPTION
**Pull request by tdammers**

### CODE REVIEW

File: cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs [hunk 1]
```diff
@@ -166,7 +160,8 @@ instance VRFAlgorithm SimpleVRF where
     let u = h' (toCBOR a) k
         y = h $ toCBOR a <> toCBOR u
         VerKeySimpleVRF v = deriveVerKeyVRF sk
-    r <- getR
+
+    let r = fromIntegral (bsToNat y) `mod` q
```
Comment (dcoutts) [inline cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs:164]: While we're at it, can we add an extra patch to replace all uses of `bsToNat` by `bytesToNatural` from the `Util` module, since that one has an efficient implementation, and tests.

File: cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs [hunk 2]
```diff
@@ -111,7 +109,7 @@ class ( Typeable v
   type Signable c = Empty
 
   evalVRF
-    :: (MonadRandom m, HasCallStack, Signable v a)
+    :: (Monad m, HasCallStack, Signable v a)
```
Comment (dcoutts) [inline cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs:112]: Can we eliminate `Monad` and just make it pure? (and hence same for `evalCertified`)

File: cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs [hunk 3]
```diff
@@ -52,7 +49,7 @@ q :: Integer
 q = C.ecc_n $ C.common_curve curve
 
 newtype Point = ThunkyPoint C.Point
-  deriving (Eq, Generic)
+  deriving stock (Eq, Generic)
```
Comment (dcoutts) [inline cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs:52]: Not bad, but unrelated.

If we wanted to do this consistently, we should do it consistently in a separate patch (and perhaps separate PR).

Comment (tdammers) [reply cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs:52]: Was just trying to shut up the warnings. But should be done separately, yes.

File: cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs (hunk 2 above)
Comment (tdammers) [reply cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs:112]: Well, we could demand that it be entirely pure, but then we can't error out anymore.

Comment (tdammers) [reply cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs:112]: Unless we throw from pure code, of course.

Comment (tdammers) [reply cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs]: OK, done. The Praos impl now just uses `error` to fail when it finds invalid keys.

File: cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs (hunk 1 above)
Comment (tdammers) [reply cardano-crypto-class/src/Cardano/Crypto/VRF/Simple.hs]: done!

File: cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs [hunk 4]
```diff
@@ -429,12 +429,12 @@ instance VRFAlgorithm PraosVRF where
...
-  evalVRF = \_ msg (SignKeyPraosVRF sk) -> do
+  evalVRF = \_ msg (SignKeyPraosVRF sk) ->
     let msgBS = serialize' msg
-    proof <- maybe (error "Invalid Key") pure $ prove sk msgBS
-    output <- maybe (error "Invalid Proof") pure $ outputFromProof proof
-    return $ output `seq` proof `seq`
-             (OutputVRF (outputBytes output), CertPraosVRF proof)
+        proof = maybe (error "Invalid Key") id $ prove sk msgBS
+        output = maybe (error "Invalid Proof") id $ outputFromProof proof
+    in output `seq` proof `seq`
```
Comment (dcoutts) [inline cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs:436]: Unrelated to this PR, but we should prefer bang patterns on the let bound vars to the use of seq.

**Comment by nc6** [comment]
But note there are a bunch of warnings that need fixing!
